File: 0392-is-subsequence/0392-is-subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        LEFT_BOUND = len(s)
        RIGHT_BOUND = len(t)

        # Devide and Conquer
        def rec_isSubsequence(left_index, right_index):
            if left_index == LEFT_BOUND:
                return True
            if right_index == RIGHT_BOUND:
                return False
            if s[left_index] == t[right_index]:
                left_index += 1
            right_index += 1

            return rec_isSubsequence(left_index, right_index)

        return rec_isSubsequence(0, 0)

File: 0392-is-subsequence/0392-is-subsequence.java
class Solution {
    String source, target;
    Integer leftBound, rightBound;

    private boolean rec_isSubsequence(int leftIndex, int rightIndex) {
        if (leftIndex == leftBound)
            return true;
        if (rightIndex == rightBound)
            return false;

        if (source.charAt(leftIndex) == target.charAt(rightIndex))
            leftIndex++;
        rightIndex++;

        return rec_isSubsequence(leftIndex, rightIndex);
    }

    public boolean isSubsequence(String s, String t) {
        this.source = s;
        this.target = t;
        this.leftBound = s.length();
        this.rightBound = t.length();

        return rec_isSubsequence(0, 0);
    }
}